src/configuration/config.contract.type.checker.ts

```diff
@@ -0,0 +1,87 @@
+import { Config } from './types';
+import {
+  ConfigurationContract,
+  ConfigurationProperty,
+} from './config.contract';
+import { ConfigContractLoader } from './config.contract.loader';
+
+export class ConfigContractTypeChecker {
+  constructor(private configContractLoader: ConfigContractLoader) {}
+
+  public checkContractTypeCompatibility(
+    configContractPath: string,
+    config: Config,
+  ) {
+    const configContract =
+      this.configContractLoader.loadConfigFormatFile(configContractPath);
+    this.compareConfigContractAndConfig(configContract, config);
+  }
+
+  private compareConfigContractAndConfig(
+    configContract: ConfigurationContract,
+    config: Config,
+  ) {
+    Object.keys(configContract).forEach((propertyName) => {
+      const contractProperty = configContract[propertyName];
+      const configProperty = config[propertyName];
+
+      if (this.isConfigProperty(contractProperty)) {
+        if (!configProperty) {
+          if (
+            !this.isContractPropertyOptional(
+              contractProperty as ConfigurationProperty,
+            )
+          ) {
+            throw new Error(
+              `The property ${propertyName} of your configuration contract is missing in your config file.`,
```

Comment:
Remplace `in your config file` par `in you configuration values`. Les valeurs peuvent venir de l'api comme d'un fichier.